src/third_party/geospatial/geospatial_aggr.cpp

```diff
@@ -11,30 +13,159 @@
 namespace TQm {
 namespace TStreamAggrs {
 
+/////////////////////////////
+///TGPSMeasurement
+TGPSMeasurement::TGPSMeasurement() {
+    //Json = TJsonVal::NewObj();
+    //Json->PutNull();
+}
+
+PJsonVal TGPSMeasurement::ToJson() const {
+    //if (Json->IsNull()) {
+    PJsonVal Json = TJsonVal::NewObj();
+    Json->AddToObj("time", Time);
+    Json->AddToObj("latitude", LatLon.Lat);
+    Json->AddToObj("longitude", LatLon.Lon);
+    Json->AddToObj("accuracy", Accuracy);
+    Json->AddToObj("speed", Speed);
+    Json->AddToObj("distanceDiff", Distance);
+    Json->AddToObj("timeDiff", TimeDiff);
+    //}
+    return Json;
+}
+
+
 ///////////////////////////////
-// StayPoint detector aggregate
-void TStayPointDetector::OnAddRec(const TRec& Rec, const TWPt<TStreamAggr>& CallerAggr) {
-    TScopeStopWatch StopWatch(ExeTm);
-    // TODO logic
-    State = TJsonVal::NewObj();
-    State->AddToObj("status", 0);
-    // dump parameters
-    State->AddToObj("parameters", Params);
-    // copy record
-    TTm Timestamp; Rec.GetFieldTm(TimeFieldId, Timestamp);
-    State->AddToObj("lastTimestamp",
-        (double)TTm::GetUnixMSecsFromWinMSecs(Timestamp.GetMSecsFromTm(Timestamp)));
-    State->AddToObj("lastLocationX", Rec.GetFieldFltPr(LocationFieldId).Val1);
-    State->AddToObj("lastLocationY", Rec.GetFieldFltPr(LocationFieldId).Val2);
-    State->AddToObj("lastAccuracy", Rec.GetFieldByte(AccuracyFieldId));
+/// TStaypointCluster
+void TGeoCluster::AddPoint(const TInt Idx,
```

Comment:
Changing this. I did a very quick search and it's not clear to me (and some other people) whether passing int as a reference makes the code slower: http://stackoverflow.com/questions/7701221/how-to-pass-literal-integers-by-reference-in-c-newbie